### DISCUSSION TITLE
Is it possible to use jotai with swr

### DISCUSSION THREAD

**Discussion by nanokind** [pmndrs/jotai#2126] [Q&A] [answered]
[SWR](https://swr.vercel.app/) export a React hook named `useSWR`. I am curious that is it possible to use `jotai` with `swr` or we need to make a feature request in `swr` repo to export core API other than hook API.

I'd appreciate it if anyone could help answer it.

**Comment by dai-shi** [comment, marked as answer]
It depends on your requirement.
Jotai has `useAtom` hook and you can create a custom hook with `useSWR`.

On the other hand, Jotai atoms have nothing to do with React and they are framework-agnostic. Unless SWR provides non React API, there are no chances.
`jotai-tanstack-query` is developed with TanStack Query core which is independent from React.

**Reply by nanokind** [reply]
Wow appreciate for your responsive reply. In my case, I'd like to use `swr` to create some async atoms, and add derived atoms from them in module level. Maybe I should consider switch to TanStack Query as it's officially supported. Thanks!
